src/api/requestsApi.tsx

```diff
@@ -174,10 +202,63 @@ const getPagedRequests = async (
   const requestedFields =
     "Id,positionTitle,requestType,paySystem,series,grade,officeSymbol,stage,subStage,Created," +
     "Author/Id,Author/EMail,Author/Title";
-
   const expandedFields = "Author";
 
-  let queryString = "ContentType eq 'RPADocSet'";
+  let queryString =
+    "ContentType eq 'RPADocSet' and stage ne 'Complete'" +
+    " and stage ne 'Cancelled'";
+
+  // if not showing all open requests, filter for those items relevant to current user
+  // current user may have multiple roles!
+  if (!allOpen) {
+    // if current user is a CSF or COSF, they may see all post-draft requests
+    // no other filters need be added
+    if (myRoles.isCSF || myRoles.isCOSF) {
+      queryString += ` and (stage ne 'Draft' or Author/EMail eq '${_spPageContextInfo.userEmail}')`;
+    } else {
+      //current user is not CSF or OSF, apply additional filters
+      const roleFilters = [];
+
+      // if you're the author, you can always view your items
+      roleFilters.push(`Author/EMail eq '${_spPageContextInfo.userEmail}'`);
+
+      // HRL sees requests assigned to them, and unassigned requests
+      if (myRoles.isHRL) {
+        roleFilters.push(
+          `stage ne 'Draft' and (hrl/Id eq null or hrl/EMail eq '${_spPageContextInfo.userEmail}')`
+        );
+      }
+
+      // OSF sees requests assigned to their org
+      if (myRoles.isOSF) {
+        const myOrgs = OSFs.filter((osf) => {
+          return osf.email === _spPageContextInfo.userEmail;
+        });
+        myOrgs.forEach((org) => {
+          roleFilters.push(`stage ne 'Draft' and (osf eq '${org.Title}')`);
+        });
+      }
+
+      // HQ only see NH-04 requests in the Candidate Offer Stage
+      if (myRoles.isHQ) {
+        roleFilters.push(
+          `stage eq 'PackageApproval' and requestType eq 'Create/Update NH-04')`
+        );
+      }
+
+      if (roleFilters.length >= 1) {
+        queryString += ` and (`;
+        roleFilters.forEach((filter, index) => {
+          if (index >= 1) {
+            queryString += ` or `;
```

Comment:
Might be safest to wrap the or statements in parenthesis too, as I'm not 100% sure on all the order of operations and how the different logic in the piecses will be evaluated without adding in explicit parenthesis for grouping